packages/x-charts/src/ChartsOnClickHandler/ChartsOnClickHandler.tsx

```diff
@@ -0,0 +1,116 @@
+};
+
+type AxisData = {
+  dataIndex: number;
+  axisValue?: number | Date | string;
+  seriesValues: Record<string, number | null | undefined>;
+};
+
+export type OnClickData = {
+  item: null | ItemData;
+  axis: null | AxisData;
+};
+
+export interface ChartsOnClickHandlerProps {
+  /**
+   * The function called for onCLick events.
+   * @param {MouseEvent} event The mouse event catched by `<svg/>`.
+   * @param {OnClickData} data The data linked to click with shape `{item, axis}`.
+   */
+  onClick?: (event: MouseEvent, data: OnClickData) => void;
+}
+
+function getItemData(
+  item: null | ItemInteractionData<ChartSeriesType>,
+  series: FormattedSeries,
+): null | {
+  dataIndex?: number;
+  series: ChartSeriesDefaultized<ChartSeriesType>;
+} {
+  if (item === null) {
+    return null;
+  }
+  return { series: series[item.type]?.series[item.seriesId]!, dataIndex: item.dataIndex };
+}
+
+function ChartsOnClickHandler(props: ChartsOnClickHandlerProps) {
+  const { onClick } = props;
+
+  const svgRef = React.useContext(SVGContext);
+  const series = React.useContext(SeriesContext);
+  const { item, axis } = React.useContext(InteractionContext);
+  const { xAxisIds, xAxis, yAxisIds, yAxis } = React.useContext(CartesianContext);
+
+  React.useEffect(() => {
+    const element = svgRef.current;
+    if (element === null || !onClick) {
+      return () => {};
+    }
+
+    const handleMouseClick = (event: MouseEvent) => {
+      event.preventDefault();
+
+      // Manage item data
+      const itemData = getItemData(item, series);
+
+      // Manage axis data
+      const isXaxis = (axis.x && axis.x.index) !== undefined;
+      const USED_AXIS_ID = isXaxis ? xAxisIds[0] : yAxisIds[0];
+      const dataIndex = isXaxis ? axis.x && axis.x.index : axis.y && axis.y.index;
+
+      if (dataIndex == null) {
+        onClick(event, { item: itemData, axis: null });
+        return;
+      }
+
+      const seriesValues: Record<string, number | null | undefined> = {};
+
+      Object.keys(series)
+        .filter((seriesType): seriesType is 'bar' | 'line' => ['bar', 'line'].includes(seriesType))
+        .forEach((seriesType) => {
+          series[seriesType]?.seriesOrder.forEach((seriesId) => {
+            const seriesItem = series[seriesType]!.series[seriesId];
+            const axisKey = isXaxis ? seriesItem.xAxisKey : seriesItem.yAxisKey;
+            if (axisKey === undefined || axisKey === USED_AXIS_ID) {
+              seriesValues[seriesId] = seriesItem.data[dataIndex];
+            }
+          });
+        });
+      const axisValue = (isXaxis ? xAxis : yAxis)[USED_AXIS_ID].data?.[dataIndex];
+      onClick(event, { item: itemData, axis: { dataIndex, axisValue, seriesValues } });
+    };
+
+    element.addEventListener('click', handleMouseClick);
+    return () => {
+      element.removeEventListener('click', handleMouseClick);
+    };
+  }, [axis, item, series, svgRef, xAxisIds, yAxisIds, xAxis, yAxis, onClick]);
+
+  return <g />;
```

Comment:
What is the goal of the `<g />` element?

Reply:
One script fails without. If I remember correctly it's proptypes that are not generated

Reply:
If you return `null` `PropTypes` are not generated?

That's weird :grimacing:

Reply:
The empty fragment works :+1: